src/abstractdataframe/reshape.jl

```diff
@@ -382,14 +392,19 @@ function Base.getindex(v::StackedVector,i::Real)
     v.components[j][k]
 end
 
-function Base.getindex(v::StackedVector,i::AbstractVector{I}) where I<:Real
+function Base.getindex(v::StackedVector,i::AbstractVector)
     result = similar(v.components[1], length(i))
     for idx in 1:length(i)
         result[idx] = v[i[idx]]
     end
     result
 end
 
+function Base.getindex(v::StackedVector, i::AbstractVector{Bool})
+    length(v) == length(i) || throw(BoundsError(v, idx))
+    v[findall(i)]
```

Comment:
Would `[v[x] for x in i if x]` be fast enough? It would avoid an allocation.

I'm also surprised there is no fallback doing the same thing already. Same for `getindex(v::RepeatedVector,i::AbstractVector)`.

Reply:
OK - let us try removing them.